workers/unity/Assets/Gdk/Core/Worker/WorkerBase.cs

```diff
@@ -45,22 +41,47 @@ public bool Connect(ConnectionConfig config)
                 Connection = ConnectionUtility.LocatorConnectToSpatial((LocatorConfig) config, GetWorkerType);
             }
 
-            if (Connection == null)
+            if (Connection == null || !Connection.IsConnected)
             {
                 return false;
             }
 
+            World = new World(WorkerId);
```

Comment:
do we want to do that? what if the user has some systems running independent from any spatial os component?
that was one of the main reasons to initialise everything in the constructor

the other one is to ensure that the worker is in a valid state from the very beginning.

Reply:
I am not sure creating the world in the constructor is the best approach because it would make the logic to dispose and recreate it much harder. Also currently the process of initializing the worker and connecting it is all done synchronously in one tick so even if its not created in the constructor no ticks will be lost. We could move this to the start of the Worker.Connect call but I dont think this will make a big difference.

Reply:
Well imagine you didn't want to connect immediately on play, this change makes it so that everything being initialised during the connection attempt. Ideally we've have the basic initialisation pre-connect and then Spatial-y things on connect then clean up Spatial-y things on disconnect.

I don't think the current structure really supports that particularly well though, unless the user wants to do all pre-connection stuff in GameObjects (which is likely) but there's no guarantee of that.

Reply:
That makes sense, currently we don't have good support for non Spatial-y things because when I clear the systems I disable all of them. Currently there are many unanswered questions on dealing with the world when the scene changes etc so a lot of this is out of the scope for this ticket.

But I think separating it into a PreConnect and Connect function might work.

Reply:
Why not leave it in the constructor? :)

Reply:
My thought was that the user should be allowed to call Worker.Clear() and then after some time be able to do Worker.Connect() again. If I initialize it in the constructor, then clean everything out then doing Worker.Connect() will throw exceptions. I think that there should be a way for the user to recreate the world and all dependencies for the connection without having to reinitialize the worker.